### PR TITLE
Left/Right Input scheme, and new event system.

### PR DESCRIPTION
**Pull request by salberico**
#4

### CODE REVIEW

File: Assets/Scripts/RedRunner/Characters/CharacterInput.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
```
Comment (WilsonCWu) [inline Assets/Scripts/RedRunner/Characters/CharacterInput.cs]: should this file still be included?

File: Assets/Scripts/RedRunner/Characters/RedCharacter.cs [hunk 2]
```diff
@@ -293,8 +299,36 @@ void Awake ()
...
-		}
-		
+
+            m_RightEvent.RegisterAction(RightEvent);
+            m_LeftEvent.RegisterAction(LeftEvent);
+        }
+        
+        private void LeftEvent()
+        {
+            Debug.Log("LEFT");
```
Comment (WilsonCWu) [inline Assets/Scripts/RedRunner/Characters/RedCharacter.cs]: delete these debug statements

File: Assets/Scripts/RedRunner/Characters/RedCharacter.cs [hunk 3]
```diff
@@ -293,8 +299,36 @@ void Awake ()
...
+                Jump();
+            }
+            else
+            {
+                m_State = CharacterState.Left;
+            }
+        }
+        private void RightEvent()
+        {
+            Debug.Log("Right");
```
Comment (WilsonCWu) [inline Assets/Scripts/RedRunner/Characters/RedCharacter.cs]: this too

File: Assets/Scripts/RedRunner/Characters/RedCharacter.cs [hunk 4]
```diff
@@ -303,7 +337,36 @@ public override void OnStartLocalPlayer()
...
+            {
+                case CharacterState.Left:
+                    Move(-1f);
+                    break;
+
+                case CharacterState.Right:
+                    Move(1f);
+                    break;
+
+                default:
```
Comment (WilsonCWu) [inline Assets/Scripts/RedRunner/Characters/RedCharacter.cs:365]: document this case

File: Assets/Scripts/RedRunner/UI/RegionClick.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+public class RegionClick : MonoBehaviour, IPointerDownHandler
+{
+    [SerializeField]
+    private UnityEvent onClick = new UnityEvent();
+
+    public void OnPointerDown(PointerEventData eventData)
+    {
+        onClick.Invoke();
+    }
+}
```
Comment (WilsonCWu) [inline Assets/Scripts/RedRunner/UI/RegionClick.cs]: new line

**Review by WilsonCWu — COMMENTED** [review]
spacing mighht also have to be fixed :(

File: Assets/Scripts/RedRunner/Characters/CharacterInput.cs (hunk 1 above)
Comment (salberico) [reply Assets/Scripts/RedRunner/Characters/CharacterInput.cs]: Nope, will remove.

File: Assets/Scripts/RedRunner/Characters/RedCharacter.cs (hunk 4 above)
Comment (salberico) [reply Assets/Scripts/RedRunner/Characters/RedCharacter.cs:365]: done

**Review by WilsonCWu — APPROVED** [review]
lgtm

**Pull request merged**